src/compiler/factory.ts

```diff
@@ -1260,11 +1268,11 @@ namespace ts {
                     foundUseStrict = true;
                 }
 
-                target.push(source[i]);
+                target.unshift(source[i]);

```

Comment:
Right, but if you have existing 3 prologues, they'll be reversed.

```ts
var x = [];
[1,2,3].forEach(e => x.unshift(e))
```

`x` will have `[3, 2, 1]`.

This function should only be getting called when adding to the beginning of an empty array anyway, so just add an assertion if that's your concern.